automation/taskcluster/decision_task_nightly.py

```diff
@@ -3,95 +3,161 @@
+
+    return taskcluster.slugId(), BUILDER.build_push_task(
+        signing_task_id,
+        name="(Reference Browser) Push task",
+        description="Upload signed release builds of Reference Browser to Google Play",
+        apks=artifacts,
+        scopes=[
+            "project:mobile:focus:releng:googleplay:product:reference-browser{}".format(
+                ':dep' if is_staging else '')
         ],
-        "requires": "all-completed",
-        "payload": {
-            "features": {
-                'taskclusterProxy': True
-            },
-            "maxRunTime": 7200,
-            "image": "mozillamobile/android-components:1.9",
-            "command": [
-                "/bin/bash",
-                "--login",
-                "-cx",
-                "cd .. && git clone https://github.com/mozilla-mobile/reference-browser.git && cd reference-browser && %s" % (command)
-            ],
-            "artifacts": {
-                "public": {
-                    "type": "directory",
-                    "path": "/build/reference-browser/release",
-                    "expires": taskcluster.stringDate(expires)
-                }
-            },
-            "env": {
-                "TASK_GROUP_ID": TASK_ID
-            }
-        },
-        "provisionerId": "aws-provisioner-v1",
-        "metadata": {
-            "name": "build",
-            "description": "Building reference browser nightly",
-            "owner": "skaspari@mozilla.com",
-            "source": "https://github.com/mozilla-mobile/android-components"
-        }
-    }
+        commit=commit,
+        is_staging=is_staging
+    )
+
 
-def nightly():
+def populate_chain_of_trust_required_but_unused_files():
+    # These files are needed to keep chainOfTrust happy. However, they have no need for Reference Browser
+    # at the moment. For more details, see: https://github.com/mozilla-releng/scriptworker/pull/209/files#r184180585
+
+    for file_name in ('actions.json', 'parameters.yml'):
+        with open(file_name, 'w') as f:
+            json.dump({}, f)
+
+
+def nightly(apks, commit, is_staging):
     queue = taskcluster.Queue({'baseUrl': 'http://taskcluster/queue/v1'})
 
     task_graph = {}
-    build_task_id = taskcluster.slugId()
-    build_task = generate_build_task()
+
+    build_task_id, build_task = generate_build_task(apks)
     lib.tasks.schedule_task(queue, build_task_id, build_task)
 
     task_graph[build_task_id] = {}
-    task_graph[build_task_id]["task"] = queue.task(build_task_id)
+    task_graph[build_task_id]['task'] = queue.task(build_task_id)
+
+    sign_task_id, sign_task = generate_signing_task(build_task_id, apks, is_staging)
+    lib.tasks.schedule_task(queue, sign_task_id, sign_task)
+
+    task_graph[sign_task_id] = {}
+    task_graph[sign_task_id]['task'] = queue.task(sign_task_id)
+
+    push_task_id, push_task = generate_push_task(sign_task_id, apks, commit, is_staging)
```

Comment:
Let's not have the push task yet. Just the signing one. This way, we can finish setting up Google Play - we require an APK signed with the production key to register the product. 

We can land the push task in a follow up PR.